qadence/backends/braket/backend.py

```diff
@@ -41,21 +41,38 @@ def promote_parameters(parameters: dict[str, Tensor | float]) -> dict[str, float
     return float_params
 
 
-@dataclass(frozen=True, eq=True)
+@dataclass(eq=True)
 class Backend(BackendInterface):
     # set standard interface parameters
-    name: BackendName = BackendName.BRAKET
-    supports_ad: bool = False
-    support_bp: bool = False
-    is_remote: bool = False
-    with_measurements: bool = True
-    with_noise: bool = False
-    native_endianness: Endianness = Endianness.BIG
-    config: Configuration = Configuration()
-
-    # braket specifics
-    # TODO: include it in the configuration?
-    _device: LocalSimulator = LocalSimulator()
+
+    def __init__(
+        self,
+        name: BackendName = BackendName.BRAKET,
+        supports_ad: bool = False,
+        support_bp: bool = False,
+        is_remote: bool = False,
+        with_measurements: bool = True,
+        with_noise: bool = False,
+        native_endianness: Endianness = Endianness.BIG,
+        config: Configuration | None = None,
+    ):
+        self.name = name
+        self.supports_ad = supports_ad
+        self.support_bp = support_bp
+        self.is_remote = is_remote
+        self.with_measurements = with_measurements
+        self.with_noise = with_noise
+        self.native_endianness = native_endianness
+        if config is None:
+            self.config = Configuration()
+        elif isinstance(config, dict):
+            self.config = Configuration(**config)
+        elif isinstance(config, Configuration):
+            self.config = config
```

Comment:
I guess we could do this in the parent class, call super, and keep this. 
This fixes the default_dict issue in python 3.11, it requried removing frozen (can investigate), and then in the past it was possible to pass config as a dict, hence the isinstance dict

Reply:
Instead of this I added `field(default_config=Configuration)` everywhere. are we ok with that?